### PR TITLE
Add Package-Level Docs for Peer Recovery

### PR DESCRIPTION
**Pull request by original-brownbear**
Adding package level documentation for the peer recovery process.

NOTE: Admittedly I'm not a complete expert on this and it could be worded in unfortunate ways here and there but I figured I'd give this a shot since we don't really have all the steps for the recovery documented in a central place which makes it hard to ramp up on the internals of this functionality. Tried to keep it in the style of the snapshot + repo package level docs.

### CODE REVIEW

**Review by dnhatn — APPROVED** [review]
I like it, LGTM. Thanks Armin.

File: server/src/main/java/org/elasticsearch/indices/recovery/package-info.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+ * {@link org.elasticsearch.index.seqno.RetentionLease} functionality. More information about this functionality can be found in the
+ * {@link org.elasticsearch.index.seqno} package and the "History retention" section in the docs.
+ *
+ * <h2>Recovery Types</h2>
+ *
+ * <h3>1. Peer Recovery</h3>
+ *
+ * Peer recovery is the process of bringing a shard copy on one node, referred to as the target node below, in-sync with the shard copy on
+ * another node, referred to as the source node below. It is always the primary node of a shard that serves as the source of the recovery.
+ * On a high level, recovery happens by a combination of comparing and subsequently synchronizing files and operations from the target to
```
[the remark this answers is not in the document]
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/indices/recovery/package-info.java]: right :) fixed

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2 (obviously unrelated watcher thingy)

**Review by tlrx — COMMENTED** [review]
Nice work, thanks Armin!

**Comment by original-brownbear** [comment]
Thanks Nhat and Tanguy!

**Pull request merged**